src/runtime/SmallValue.h

```diff
@@ -77,7 +77,7 @@ const int kApiInt64Size = sizeof(int64_t);
 
 // Tag information for HeapObject.
 const intptr_t kHeapObjectTag = 0;
-const int kHeapObjectTagSize = 2;
+const int kHeapObjectTagSize = 1;
```

Comment:
Is there any reason to reduce the value of `kHeapObjectTagSize` to 1? `kHeapObjectTagSize` is used to check if SmallValue has an address of Object(PointerValue), so the two least significant bits should be 0 for both 32 bit and 64 bit platform.

Reply:
The motivation is that if the two last bit is zero, that also includes that the last bit is zero as well, and that is enough to make the difference between integer and pointer defines. Since the Value constants sets the second bit, they will not match the Object checks currently used now. This part could be redesigned later, but I would not do it in this patch to make it less risky.

Reply:
I think so. Object check (`HAS_OBJECT_TAG` may also?) needs to be modified properly later.